runtime/lua/vim/lsp/protocol.lua

```diff
@@ -744,14 +744,15 @@ function protocol.make_client_capabilities()
       completion = {
         dynamicRegistration = false,
         completionItem = {
-          -- Until we can actually expand snippet, move cursor and allow for true snippet experience,
-          -- this should be disabled out of the box.
-          -- However, users can turn this back on if they have a snippet plugin.
-          snippetSupport = false,
+          snippetSupport = true,
```

Comment:
If we do this (which I agree we should), we should set them up only within the snippet context and tear them down afterwards (to avoid conflicting with snippet _library_ plugins). Would be easier with namespaced mappings, of course, but... I don't think the overhead for this would be relevant.

(Note for the sake of clarity: This suggestion is -- for now -- only for the MVP, which should be as hermetic as possible and only focus on out-of-the-box LSP autocompletion. How to expose parts of this as an API for other plugins to extend is a follow-up discussion after the bugs are ironed out and we have a complete picture; we should make use of the full 0.11 dev cycle to iterate on this.)

Reply:
Done in https://github.com/neovim/neovim/pull/27339/commits/1fa63b786f7c66b2fe90dff69d6d5338e2cd841d.
@clason lmk if this is what you had in mind.